source/common/router/router.h

```diff
@@ -297,14 +297,18 @@ using UpstreamRequestPtr = std::unique_ptr<UpstreamRequest>;
  */
 class Filter : Logger::Loggable<Logger::Id::router>,
                public Http::StreamDecoderFilter,
+               public Http::UpstreamCallbacks,
                public Upstream::LoadBalancerContextBase,
                public RouterFilterInterface {
 public:
   Filter(FilterConfig& config, FilterStats& stats)
       : config_(config), stats_(stats), grpc_request_(false), exclude_http_code_stats_(false),
         downstream_response_started_(false), downstream_end_stream_(false), is_retry_(false),
-        request_buffer_overflowed_(false), streaming_shadows_(Runtime::runtimeFeatureEnabled(
-                                               "envoy.reloadable_features.streaming_shadow")) {}
+        ensure_connection_retry_(
+            Runtime::runtimeFeatureEnabled("envoy.restart_features.ensure_connection_retry")),
+        wait_for_connect_(false), request_buffer_overflowed_(false),
```

Comment:
why doesn't wait_for_connect_ start out as true?

Reply:
I guess it make more sense to keep it disable until we create the 1st upstream make sure we have the conditions to wait for a connection.
But I don't have a strong opinion, I can also default it to true and disable it if we do not ensure connection retry